crates/host/src/wasmbus/mod.rs

```diff
@@ -747,6 +749,66 @@ impl Messaging for Handler {
 }
 
 impl ActorInstance {
+    #[instrument(skip(self))]
+    async fn handle_invocation(
+        &self,
+        contract_id: &str,
+        operation: &str,
+        msg: Vec<u8>,
+    ) -> anyhow::Result<Result<Vec<u8>, String>> {
+        let mut instance = self
+            .pool
+            .instantiate(self.runtime.clone())
+            .await
+            .context("failed to instantiate actor")?;
+        instance
+            .stderr(stderr())
+            .await
+            .context("failed to set stderr")?
+            .bus(Arc::new(self.handler.clone()))
+            .keyvalue_readwrite(Arc::new(self.handler.clone()))
+            .messaging(Arc::new(self.handler.clone()));
+        #[allow(clippy::single_match_else)] // TODO: Remove once more interfaces supported
+        match (contract_id, operation) {
+            ("wasmcloud:httpserver", "HttpServer.HandleRequest") => {
+                let req: wasmcloud_compat::HttpRequest =
+                    rmp_serde::from_slice(&msg).context("failed to decode HTTP request")?;
+                let req = http::Request::try_from(req).context("failed to convert request")?;
+                let res = match wasmcloud_runtime::ActorInstance::from(instance)
+                    .into_incoming_http()
+                    .await
+                    .context("failed to instantiate `wasi:http/incoming-handler`")?
+                    .handle(req.map(|body| -> Box<dyn AsyncRead + Send + Sync + Unpin> {
+                        Box::new(Cursor::new(body))
+                    }))
+                    .await
+                {
+                    Ok(res) => res,
+                    Err(err) => return Ok(Err(format!("{err:#}"))),
```

Comment:
Is this for pretty printing?

Reply:
This is how errors are printed throughout runtime crate, which provides the error chain

Reply:
See https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations